samcli/local/apigw/local_apigw_service.py

```diff
@@ -143,7 +155,7 @@ def _request_handler(self, **kwargs):
         route = self._get_current_route(request)
 
         try:
-            event = self._construct_event(request, self.port, route.binary_types)
+            event = self._construct_event(request, self.port, route.binary_types, route)
```

Comment:
Don't pass the whole route. Extract the information you need from it, like how we have done for `binary_types`.